extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergCatalog.java

```diff
@@ -0,0 +1,221 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.iceberg.util;
+
+import io.deephaven.base.verify.Assert;
+import io.deephaven.engine.context.ExecutionContext;
+import io.deephaven.engine.table.Table;
+import io.deephaven.engine.table.TableDefinition;
+import io.deephaven.engine.table.impl.PartitionAwareSourceTable;
+import io.deephaven.engine.table.impl.locations.impl.PollingTableLocationProvider;
+import io.deephaven.engine.table.impl.locations.impl.StandaloneTableKey;
+import io.deephaven.engine.table.impl.locations.impl.TableLocationKeyFinder;
+import io.deephaven.engine.table.impl.locations.util.TableDataRefreshService;
+import io.deephaven.engine.table.impl.sources.regioned.RegionedTableComponentFactoryImpl;
+import io.deephaven.engine.updategraph.UpdateSourceRegistrar;
+import io.deephaven.extensions.s3.Credentials;
+import io.deephaven.extensions.s3.S3Instructions;
+import io.deephaven.iceberg.layout.IcebergFlatLayout;
+import io.deephaven.iceberg.layout.IcebergPartitionedLayout;
+import io.deephaven.iceberg.location.IcebergTableLocationFactory;
+import io.deephaven.iceberg.location.IcebergTableLocationKey;
+import io.deephaven.parquet.table.ParquetInstructions;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.iceberg.*;
+import org.apache.iceberg.aws.AwsClientProperties;
+import org.apache.iceberg.aws.s3.S3FileIOProperties;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.rest.RESTCatalog;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class IcebergCatalog {
+    private final Catalog catalog;
+    private final FileIO fileIO;
+
+    private final S3Instructions s3Instructions;
+
+    /**
+     * Construct an IcebergCatalog given a set of configurable instructions..
+     *
+     * @param name The optional service name
+     */
+    IcebergCatalog(final @Nullable String name, final IcebergInstructions instructions) {
+        // Set up the properties map for the Iceberg catalog
+        Map<String, String> properties = new HashMap<>();
```

Comment:
Nitpick:
```suggestion
        final Map<String, String> properties = new HashMap<>();
```

You can turn on suggestions in IntelliJ to point out all variables which can be made final.